### DISCUSSION TITLE
Missing slider ticks needs to impact accuracy more

### DISCUSSION THREAD

**Discussion by Sleepteiner** [ppy/osu#26552] [Ideas]
It is good that slider ends have more weight on the score now, but slider ticks need the same increased weight treatment too. Since the scoring system is much more accuracy focused than ever before, which is a very good thing, accuracy also needs to make more sense than ever before as well if we want the scoring system to be accurate itself. Accuracy needs to roughly reflect how well the player is playing the map. Slider breaking on one tick of a two tick slider and still getting 94.11% accuracy from that whole slider makes no sense.

I played [Krzysztof Krawczyk - Bo jestes ty [Banalny]](https://osu.ppy.sh/beatmapsets/371675#osu/814198), which is a ranked map made entirely of sliders with slider ticks (and two spinners), while intentionally missing every single slider tick. I was still awarded with a B rank at 81.23% accuracy. This is ridiculous considering, if you ignore the two spinners, I didn't play a single object in this whole map correctly. This play needs to be a D rank, in my opinion.

![osu_2024-01-15_14-00-25](https://github.com/ppy/osu/assets/55634901/8d3f6dbb-70c9-47f6-b72d-436cc44a2a4b)

Having mistakes not be sufficiently reflected in accuracy in a competitive setting encourages players to intentionally allow those mistakes to happen in order to make bigger mistakes less likely. For example, if a player is trying to get an A rank on a map with hard sliders in the current system, there would absolutely be plenty of times where it increases the player's chances of an A rank to just aim from the slider head to the slider end and ignore the body of some of the harder sliders with slider ticks as that may make aiming the next circle easier. It will encourage players, especially newer players, to play maps incorrectly which will then pressure mappers to stop using those kinds of difficult sliders in their maps.

I acknowledge that if the weight on slider ticks increases, certain kinds of longer sliders will start to be weighted too much compared to circles. However, that is a fundamental problem with the whole system of having each component of a slider contribute to score and accuracy individually instead of a combined impact in a game where mappers tend to use sliders conceptually as a single object, not multiple objects. And, in my opinion, having those long map-end sliders and high bpm repeat sliders be worth too much is a lesser problem than accuracy being made less representative of a player's performance across the board.

EDIT (Jan 19): I probably should add on to this with a specific and replicable example. A couple days ago, I was playing some 1 star maps converted to CS11. I was trying to get the best accuracy that I could on them. Due to slider ticks having only a tiny impact on accuracy, the best strategy was undeniably to just completely ignore the harder slider ticks. From my perspective, encouraging players to play like that when going for good accuracy just seems like really poor game design. I really hope it can be changed before score submission goes fully online.

EDIT (Jan 30): @ssz7-ch2 generously made some more slider tick weight play comparisons for me and it turns out that slider tick weight only needs to increase from the current 30 to 56 to make the play on [Bo jestes ty](https://osu.ppy.sh/beatmapsets/371675#osu/814198) that was mentioned above into a D rank.

With a slider tick score weight of 56, that play would have an accuracy of 69.86%:

![osu_2024-01-30_19-33-38](https://github.com/ppy/osu/assets/55634901/2fbc32a7-e86a-477c-9856-dbb5ab300657)

With a slider tick score weight of 60, that play would have an accuracy of 68.39%:

![osu_2024-01-30_19-00-45](https://github.com/ppy/osu/assets/55634901/3e5b0804-252d-461d-b7cc-7034cfe28afe)

Partially because of these findings, I am now specifically advocating for increasing the slider tick weight from 30 to 75, not all the way to 150 like with slider ends. I think 75 is a good number since it is both over the 56 score weight threshold to mostly fix the results of that [Bo jestes ty](https://osu.ppy.sh/beatmapsets/371675#osu/814198) play, and also since it is exactly half of the slider end's 150 score weight which would make it easier to understand and explain from a player perspective. A slider tick would be worth half of a slider end which is worth half of a slider head.

Since one of the most common arguments against my general proposal to increase the accuracy weight of slider ticks is the belief that "almost no one would ever skip a slider tick to get better accuracy if they break combo in the process", I would like to remind everyone that in the [March 2022 Scoring Survey](https://osu.ppy.sh/home/news/2022-04-17-scoring-survey-results), 9.9% of players said that they would "primarily sort per-beatmap leaderboards" by accuracy if they had the choice. Sure, 9.9% of players is a minority, but for a game this big, that is a lot of players who prefer accuracy to score and pp as the main way of comparing plays on a map. Breaking the logic of the accuracy statistic by making slider ticks have only a tiny impact on accuracy will be yet another reason why that 9.9% of players will be hesitant to switch to lazer.

EDIT (Feb 3): I thought of another way to further convey my point. I made a practice difficulty of [Halozy - Kikoku Doukoku Jigokuraku [Notch Hell]](https://osu.ppy.sh/beatmapsets/437683#osu/942356) that includes only the hardest section of the map (4:57:913-5:39:139). I played that practice difficulty 20 times. The first time I attempted to play all of the hard sliders correctly. The second time I simply aimed from the slider head to the slider end of all of the hardest sliders. The third time I switched back to playing the map correctly and I alternated strategies from there. This means that I played the map correctly 10 times and I played the map skipping all of the hard slider ticks 10 times as well. Here are the accuracy results:

| Try to hit all slider ticks (30 score weight) | Skip hard slider ticks (30 score weight) | 
|     :---:     |     :---:     | 
| 89.98 | 93.28 |
| 94.13 | 96.07 |
| 93.38 | 96.26 |
| 94.04 | 96.95 |
| 93.54 | 95.67 |
| 95.80 | 93.64 |
| 96.26 | 95.08 |
| 93.64 | 95.90 |
| 96.85 | 95.97 |
| 98.82 | 97.34 |
| Average: **94.64** | Average: **95.62** |

Even with that lucky 98.82% outlier play at the end, the strategy of ignoring the hard slider ticks ended up with 0.98% better accuracy on average.

This accuracy difference between playing the map correctly and using a cheese strategy will only get wider and wider the less experience a player has playing difficult sliders.

It is also very important to note that I didn't get close to FCing this section on any of the attempts that I tried to. If the player is going to combo break on a section anyway, that makes accuracy the only important thing to focus on in that section. This then makes this issue larger than just having a negative impact on accuracy players. **This indicates that for any player playing a map that they can't FC where the hardest part of the map is a slider heavy section with slider ticks, the undeniable best strategy in most cases for that player if they want to improve their score efficiently no matter if they are going for accuracy, or score, or pp, is to ignore the slider ticks of the hardest sliders.**

Imagine if the best strategy for getting the best score a player can on a traditional aim map that that player can't FC is to simply ignore the hardest jumps of the map. This is the current reality of maps where the hardest part is a slider section with slider ticks.

I then recalculated the accuracy of all of those same plays on the practice diff to be what they would be if slider ticks had a score weight of 75 instead of 30. Here are the accuracy results of that:

| Try to hit all slider ticks (75 score weight) | Skip hard slider ticks (75 score weight) | 
|     :---:     |     :---:     | 
| 88.99 | 89.60 |
| 93.42 | 92.20 |
| 93.27 | 92.66 |
| 93.88 | 93.57 |
| 93.42 | 91.97 |
| 95.25 | 90.21 |
| 96.10 | 91.28 |
| 92.96 | 92.73 |
| 96.78 | 91.97 |
| 98.62 | 93.80 |
| Average: **94.27** | Average: **92.00** |

With a slider tick score weight of 75, the average accuracy of the plays that cheesed the hard sliders was 2.27% lower than the average accuracy of the plays that played the sliders correctly. And, the highest possible accuracy that I could get with the cheese strategy is still lower than the average accuracy I got with the intended strategy. This is a dramatic improvement over the current 30 slider tick score weight.

If the slider tick score weight was 75, there would no reason in most cases for the player to use the cheese strategy of ignoring all of the hard slider ticks in a map they can't FC. The game would be encouraging the player through the scoring systems to play the maps as they were intended to be played by the mapper.

To anyone reading this, if you still aren't convinced about some part of my proposal or if I need to provide even more evidence of this problem, please tell me. I am very invested in conveying the severity of this game design problem since it would be depressing for me if osu goes the direction of making gameplay changes that break certain aspects of the game, like making slider ticks have very little impact on accuracy, without anywhere close to the sufficient amount of research and consideration needed to get the full picture of the impacts of those changes.

**Comment by felipemarins** [comment]
While I think your concern may be valid, I don't know how to feel about adding more non-rhythmic accuracy factors into the accuracy percentage. The examples you provided seem to show more the grade system flaws than the score system flaws (I would say that a 106 154 score is fine for that play). Although the accuracy have more impact now in the current score system, the combo continues to be important and I think that having the slider tick misses break combo is already enough as a punishment. I don't know if getting a low combo score would be worth just to get an A grade if your actual score is low (at most, I think that the grade system would need some change).

**Reply by Sleepteiner** [reply]
> I don't know how to feel about adding more non-rhythmic accuracy factors into the accuracy percentage.

It isn't non-rhythmic. Hitting slider ticks requires rhythmic aiming.

And, this isn't adding anything new, this is just restoring a small part of something that already existed in stable in some form.

> The examples you provided seem to show more the grade system flaws than the score system flaws

I could change my example of a player going for an A rank to a player going for a 90% accuracy score and my point wouldn't change. In my opinion, grades being based on accuracy is not the problem. The problem here is the way accuracy works.

> I would say that a 106 154 score is fine for that play

I have a hard time agreeing that a play that played virtually every object completely incorrectly should be worth 100k points. 

> I don't know if getting a low combo score would be worth just to get an A grade if your actual score is low

It is worth it for people who just want the A rank and aren't concerned with the amount of points they get.

**Reply by felipemarins** [reply]
> It isn't non-rhythmic. Hitting slider ticks requires rhythmic aiming.

Yeah, to a certain point it requires some level of rhythmic precision, but the most part of it is aim accuracy rather than actual rhythmic accuracy (or just "accuracy" as defined by the wiki: "ability to hit hit objects in time").

> I have a hard time agreeing that a play that played virtually every object completely incorrectly should be worth 100k points.

Well, if you perfectly hit every hit object that give a rhythmic judgement, I think it's deserved to get 100k points. As an exemple, if someone is learning a song in the guitar and they just play the start of every note, but perfectly get the timing of every note start, I don't think it would be a good performance but it would not be so bad that it needs a score below 100k with a maximum of 1M.

> The problem here is the way accuracy works.

As it's probably clear by now, I think that rhythmic judgements (Great, Ok, and Meh, or just 300, 100, and 50 on stable) should have more weight on accuracy than aim judgements. I think it's fine for slider ends to have the current weight on accuracy as they don't break combo and most of the time it impacts the time you have to hit the next hit object and don't get a 100. But, in my opinion, hitting slider ticks is mostly dependent on aim accuracy and they already break combo so I don't know if increasing its weight on accuracy is needed. Maybe I just have a weird view on accuracy.

> It is worth it for people who just want the A rank and aren't concerned with the amount of points they get.

If they are aiming only for an achievement, I think that's true. It could be fixed with changes on the grade system (maybe turning it into a score-based system, but some people seem to really dislike this idea).

As a conclusion, I have some concerns about your proposal. Maybe it's because I'm just used to how lazer works. If you could provide an example of how you wanted that score to be, maybe I can have a better idea of the impact it would have.

**Reply by Sleepteiner** [reply]
> I think that rhythmic judgements (Great, Ok, and Meh, or just 300, 100, and 50 on stable) should have more weight on accuracy than aim judgements

I'm not suggesting individual slider ticks be worth more than circles.

> I don't know if increasing its weight on accuracy is needed.

Other than longer sliders being a bit too impactful, what problems would increasing the slider tick's weight on accuracy actually create? How would it negatively impact players?

> If you could provide an example of how you wanted that score to be, maybe I can have a better idea of the impact it would have.

I want that score to be a D rank since I didn't play any of the non-spinner objects correctly, so, in my opinion, the weight of slider ticks should be however high it needs to be for that score, and scores like it, to be below 60% accuracy. Increasing the slider tick's weight to be the same as slider ends would probably be enough.

**Reply by felipemarins** [reply]
> I'm not suggesting individual slider ticks be worth more than circles.

I guess I phrased it wrongly. My opinion is that rhythmic judgements should account for at least 80% of the accuracy part (ideally it would account for nearly 100%). As I said previously in another discussion, I think that hitting the start of a note should have much more importance than anything else. And making aim judgements account for nearly 50% of that seems wrong to me.

> Other than longer sliders being a bit too impactful, what problems would increasing the slider tick's weight on accuracy actually create? How would it negatively impact players?

Other than proportionally devaluing the rhythmic judgements, the accuracy percentage could loose one of its uses as a quick rough metric for how good you were in hitting circles/slider heads in time (especially in slider heavy maps). Because of the slider ends increased weight, it isn't as precise as it was before on lazer, but it still serves as an estimative. I like how the score impact that slider ends have now, I just think they should be in a separate stat from rhythmic judgments (something I know is impossible by now).

> I want that score to be a D rank since I didn't play any of the non-spinner objects correctly, so, in my opinion, the weight of slider ticks should be however high it needs to be for that score, and scores like it, to be below 60% accuracy.

I guess we can't really agree on this one, as I like how sliders work on lazer and I think the score is just fine.

**Reply by Sleepteiner** [reply]
> the accuracy percentage could loose one of its uses as a quick rough metric for how good you were in hitting circles/slider heads in time

We can already essentially do this by looking at the ratio of 300s to 100s+50s and the UR.

In the big picture, "rhythmic judgements" being the only important thing just isn't how the accuracy statistic has ever worked. Accuracy has always included misses and misaimed sliders and had them have significant impact. The skills related to avoiding those mistakes are what all accuracy players have always had to work to improve to get better accuracy. Trying to mold what accuracy is into something that is fundamentally different is something that I just can't be in favor of without obvious huge benefit.

I would, however, be in favor of another UR type statistic that measures how close the player is to the timing center of a map on average in a play so that there is a metric in the game that measures only the player's pure rhythm clicking ability. We could even break score ties with this if that is what leaderboard-focused players want.

**Reply by sz-da** [reply]
> I don't think it would be a good performance but it would not be so bad that it needs a score below 100k with a maximum of 1M

Note this score is set with NF so the maximum is actually 500k. And 20% of the max score is quite high for this particular score imo

**Comment by sz-da** [comment]
Providing some examples with slidertick score at 30, 60, and 100.

<details>
  <summary>Table with comparisons</summary>

  ΔAcc is between 30 and 60

  | 30 | 60 | 100 | Slidertick Ratio | ΔAcc |
  | -- | -- | -- | -- | -- |
  | ![frieren 30](https://github.com/ppy/osu/assets/76718358/9d1ad9e9-4de4-434e-9d84-23d32fa02762) | ![frieren 60](https://github.com/ppy/osu/assets/76718358/23b824e8-7c8f-40d3-8873-f8cefe0b7a58) | ![frieren 100](https://github.com/ppy/osu/assets/76718358/581bba41-908e-4d19-b32f-440e3affd032) | 5.1% | +0.01% |
  | ![promise code 30](https://github.com/ppy/osu/assets/76718358/f140c81f-c575-403b-b1ba-5009cae23ee6) | ![promise code 60](https://github.com/ppy/osu/assets/76718358/68de35da-dc08-43ff-b8ca-ac846402856a) | ![promise code 100](https://github.com/ppy/osu/assets/76718358/d7cab5aa-3bf2-4615-826e-dbc0eb217dd1) | 5.3% | +0.01% |
  | ![superorbital 30](https://github.com/ppy/osu/assets/76718358/3912f1bc-a16e-4c0d-8aa3-f2c3ad4e9046) | ![superorbital 60](https://github.com/ppy/osu/assets/76718358/b39b4a69-b431-441a-8d1c-41de1612966c) | ![superorbital 100](https://github.com/ppy/osu/assets/76718358/1862e09e-b497-443d-9bd2-83dd4a72f6a4) | 6.3% | +0.07% |
  | ![nanatsu 30](https://github.com/ppy/osu/assets/76718358/b3417083-c9ff-45f6-9610-9840270b9e4d) | ![nanatsu 60](https://github.com/ppy/osu/assets/76718358/265b0e8b-b68f-4579-a203-819b9f49de04) | ![nanatsu 100](https://github.com/ppy/osu/assets/76718358/37cb2848-e423-49c2-bbf7-a91f6272468c) | 10.7% | +0.01% |
  | ![notch hell 30](https://github.com/ppy/osu/assets/76718358/99c13d67-a398-4aab-a0a5-7337540d4a8b) | ![notch hell 60](https://github.com/ppy/osu/assets/76718358/ec6979c1-6fa6-4610-aecd-b749d31de267) | ![notch hell 100](https://github.com/ppy/osu/assets/76718358/d77c9345-f0ec-4bc9-9d96-3e28e8851e0f) | 16.2% | -0.80% |
  | ![my sweet maiden 30](https://github.com/ppy/osu/assets/76718358/4da4c01b-b3f1-45d3-bf78-81710a4d690a) | ![my sweet maiden 60](https://github.com/ppy/osu/assets/76718358/8c700876-4e0a-4b23-96e6-5f9793da7b89) | ![my sweet maiden 100](https://github.com/ppy/osu/assets/76718358/6a8835fc-ca04-44c9-b4ea-ad7f8bc9acac) | 25.6% | -0.60% | 
  | ![sendan 30](https://github.com/ppy/osu/assets/76718358/f1fda040-3dab-48f5-bfb7-ed193fc44575) | ![sendan 60](https://github.com/ppy/osu/assets/76718358/4c63fb04-68ef-4d59-9c3e-4f480a8fd354) | ![sendan 100](https://github.com/ppy/osu/assets/76718358/3cda86f9-15fa-4c0c-ad96-be2047178356) | 31.4% | +0.03% |
  | ![ui 30](https://github.com/ppy/osu/assets/76718358/446d7550-e21c-4024-b435-2261517711a6) | ![ui 60](https://github.com/ppy/osu/assets/76718358/9a90ef62-1d3e-4829-b234-62e5c033b72b) | ![ui 100](https://github.com/ppy/osu/assets/76718358/d30a1dea-de6e-46b4-bcd4-2b3fb28248e8) | 32.2% | +0.04% |
  | ![kaede 30](https://github.com/ppy/osu/assets/76718358/25be6182-e892-4765-bf67-dcda3133a27e) | ![kaede 60](https://github.com/ppy/osu/assets/76718358/ed5b35c6-b8fe-49d9-8d9c-367524fff3bb) | ![kaede 100](https://github.com/ppy/osu/assets/76718358/65433bc7-00bb-440b-836a-1dc2878e0aec) | 32.8% | +0.17% |
  | ![zeitaku 30](https://github.com/ppy/osu/assets/76718358/a97c8fcd-ca92-47ed-8240-e1803edf3fa4) | ![zeitaku 60](https://github.com/ppy/osu/assets/76718358/8e2f35c5-96cc-4382-9c60-33a3fb2db196) | ![zeitaku 100](https://github.com/ppy/osu/assets/76718358/8cc73a54-75ba-43f6-bb0e-08d6369ec3c1) | 35.6% | +0.03% |
  | ![everything will freeze 30](https://github.com/ppy/osu/assets/76718358/a36c59af-cf57-4030-ad52-175364003b66) | ![everything will freeze 60](https://github.com/ppy/osu/assets/76718358/22d74280-91be-4f40-91c3-dbdc7b93ac5b) | ![everything will freeze 100](https://github.com/ppy/osu/assets/76718358/ffdb473c-2b34-4d89-82d8-542921d50df0) | 38.5% | +0.02% |
  | ![maze 30](https://github.com/ppy/osu/assets/76718358/674ff34c-03f8-4b31-96da-df593f248a23) | ![maze 60](https://github.com/ppy/osu/assets/76718358/94989373-9760-472d-a372-1927f1d862e6) | ![maze 100](https://github.com/ppy/osu/assets/76718358/150e8728-31c4-40b8-8a82-eb180171ee29) | 67.2% | -3.80% |
  | ![z 30](https://github.com/ppy/osu/assets/76718358/71b2edaf-5e7a-45cb-965a-294670464fce) | ![z 60](https://github.com/ppy/osu/assets/76718358/feeaada1-d0ea-4b23-bd6e-1ad9415da9e3) | ![z 100](https://github.com/ppy/osu/assets/76718358/e753b083-e3c2-4dd8-b88e-36723f4ae36d) | 69.1% | +0.26% |
  | ![miiro 30](https://github.com/ppy/osu/assets/76718358/5c41a0b8-d650-4e2a-8635-e0ce9d0e14ec) | ![miiro 60](https://github.com/ppy/osu/assets/76718358/c7a707e5-6692-4f50-be86-0a5d533d4d48) | ![miiro 100](https://github.com/ppy/osu/assets/76718358/322b0fe2-a120-43d5-b20a-2faf5d3e6a41) | 75.7% | +0.65% |
  | ![ultra beginner 30](https://github.com/ppy/osu/assets/76718358/619356c9-0ebc-4fed-88bb-2f748c16c32b) | ![ultra beginner 60](https://github.com/ppy/osu/assets/76718358/a8767118-24a0-43f6-9b2e-6d410b2c428d) | ![ultra beginner 100](https://github.com/ppy/osu/assets/76718358/40ab7437-cbd8-4cc2-bc0d-0c33e2ff7ce9) | 89.8% | +0.68% |
</details>



My conclusions from these examples. Feel free to draw your own conclusions from it.

- Missing sliderticks will decrease acc more
- If you hit all the sliderticks correctly, the acc change will be minimal.
- The only case where this becomes a problem is with very easy maps that are made up of over 50% sliderticks.
- Most normal/easy maps are made up of 5-40% sliderticks, and the acc change isn't very significant
- Hard+ maps rarely have 20%+ sliderticks, and even in the cases where they do, the acc change is still minimal.

I think raising slidertick score to somewhere around 50-70 shouldn't cause any major problems while punishing players more for missing sliderticks.
